cidc_api/services/utils.py

```diff
@@ -0,0 +1,43 @@
+from functools import wraps
+
+from eve.auth import requires_auth
+from flask import current_app as app
+
+
+def resource(
+    name: str,
+    public_methods: list = [],
+    allowed_roles: list = [],
+    allowed_read_roles: list = [],
+    allowed_write_roles: list = [],
+):
+    """
+    Decorator for custom endpoints that adds and configures that endpoint
+    as a resource in the Eve API's DOMAIN config.
+
+    Only works for endpoint functions that take no arguments.
+    """
+
+    def decorator(f):
+        @wraps(f)
+        def wrapped(*args, **kwargs):
+            kwargs["resource"] = name
+            app.config["DOMAIN"][name] = {
+                "public_methods": public_methods,
+                "allowed_roles": allowed_roles,
+                "allowed_read_roles": allowed_read_roles,
+                "allowed_write_roles": allowed_write_roles,
+                "authentication": app.auth,
+            }
+
+            @requires_auth("resource")
```

Comment:
shouldn't this be like 
```suggestion
            @requires_auth(name)
```

Reply:
Eve's `requires_auth` decorator expects an "endpoint class" (either `"resource"`, `"item"`, or `"home"`) as its argument: https://github.com/pyeve/eve/blob/4b052b843f7eb1b402709d8319332a818b35c52d/eve/auth.py#L16